src/main/java/ch/admin/bar/siardsuite/ui/presenter/connection/ServerBasedDbmsConnectionPropertiesForm.java

```diff
@@ -142,27 +162,46 @@ public ServerBasedDbmsConnectionPropertiesForm(
         HBox.setMargin(jdbcUrl, new Insets(25));
         val thirdLineHBox = new HBox(jdbcUrl);
 
-        this.getChildren().addAll(
-                firstLineHBox,
-                secondLineHBox,
-                thirdLineHBox
-        );
+        if (showSchemaField && "oracle".equals(dbms.getId())) {
+            this.getChildren().addAll(
+                    firstLineHBox,
+                    secondLineHBox,
+                    schemaNameBox,
+                    thirdLineHBox
+            );
+        } else {
+            this.getChildren().addAll(
+                    firstLineHBox,
+                    secondLineHBox,
+                    thirdLineHBox
+            );
+        }
 
         formFields.add(host);
         formFields.add(port);
         formFields.add(dbName);
         formFields.add(username);
         formFields.add(password);
         formFields.add(jdbcUrl);
+        if (showSchemaField && "oracle".equals(dbms.getId())) {
+            formFields.add(schema);
+        }
 
-        connectionPropertiesSupplier = () -> ServerBasedDbmsConnectionProperties.builder()
-                .host(host.getValue())
-                .port(port.getValue())
-                .dbName(dbName.getValue())
-                .user(username.getValue())
-                .password(password.getValue())
-                .options(Optional.ofNullable(connectionOptions))
-                .build();
+        connectionPropertiesSupplier = () -> {
+            val builder = ServerBasedDbmsConnectionProperties.builder()
+                    .host(host.getValue())
+                    .port(port.getValue())
+                    .dbName(dbName.getValue())
+                    .user(username.getValue())
+                    .password(password.getValue())
+                    .options(Optional.ofNullable(connectionOptions));
+
+            if (showSchemaField && "oracle".equals(dbms.getId())) {
```

Comment:
the expression `"oracle".equals(dbms.getId()` is used three times now - extract a function for that